src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] dataArray;
+    private int size;
+
+    public ArrayList() {
+        dataArray = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growUp();
+        dataArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growUp();
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index < size && index >= 0) {
+            System.arraycopy(dataArray, index, dataArray, index + 1, size - index);
+            dataArray[index] = value;
+            size++;
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException(
+                "Index " + index + " is wrong for size " + size);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > size) {
+            growUp();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            dataArray[size] = list.get(i);
+            size++;
+        }
```

Comment:
here maybe better create just for each loop with adding elements? and than create in add() your condition that :
if (list.size() > size) {
            growUp();
        }
 other away i think that every time, after each adding of element to your list, you are widening your capacity(maybe i wrong, but i think that in this method you can remove this condition and move it to add(T value))^_^